### PR TITLE
Chart resizing now also resizes the chart legend

### PR DESCRIPTION
**Pull request by anutron**

### CODE REVIEW

File: js/Source/Charts/Chart.js [hunk 1]
```diff
@@ -236,6 +236,22 @@ Chart = new Class({
     this.chart.setSize(width, height, false);
     // redraw the backgrounds we added so they fit the new dimentions
     this._drawBackgrounds();
+
+
+    // resize the legend, which requires us to overwrite
+    // numerous width values calculated when the chart was
+    // initialized, recalculate them, then reset the options
+    // of the chart to the new sizies, destroy the legend items
```
Comment (davywentworth) [inline js/Source/Charts/Chart.js]: typo

File: js/Source/Charts/Chart.js [hunk 2]
```diff
@@ -236,6 +236,22 @@ Chart = new Class({
...
+    // and re-render it
+    delete this.options.legendItemWidth;
+    delete this.options.legendWidth;
+    this._setV2Defaults();
+    this.chart.legend.options.width = this.options.legendWidth;
+    this.chart.legend.itemStyle.width = this.options.legendItemWidth;
+    this.chart.legend.options.itemWidth = this.options.legendItemWidth;
+    this.chart.legend.maxItemWidth = this.options.legendItemWidth;
+    this.chart.series.each(this.chart.legend.destroyItem, this);
+    this.chart.legend.render();
```
Comment (davywentworth) [inline js/Source/Charts/Chart.js]: isn't it slow to re-render this?

**Comment by davywentworth** [comment]
this seems fine, maybe a little slow.

File: js/Source/Charts/Chart.js (hunk 1 above)
Comment (anutron) [inline js/Source/Charts/Chart.js]: No no. Sizies is what I meant.

File: js/Source/Charts/Chart.js (hunk 2 above)
Comment (anutron) [inline js/Source/Charts/Chart.js]: nah. It's instantaneous. It basically destroys some DOM elements and recreates them.

Comment (anutron) [inline js/Source/Charts/Chart.js]: Keep in mind the legend never has more than like a dozen things.

**Pull request closed without merging**